udf/udf_get_key.sql

```diff
@@ -1,8 +1,38 @@
-CREATE TEMP FUNCTION udf_get_key(map ANY TYPE, k ANY TYPE) AS (
- (
-   SELECT key_value.value
-   FROM UNNEST(map.key_value) AS key_value
-   WHERE key_value.key = k
-   LIMIT 1
- )
-);
+/*
+
+Return the first value for a given key in a map.
+
+In BigQuery the schema for a map is STRUCT<key_value ARRAY<STRUCT<key ANY TYPE, value ANY TYPE>>>.
```

Comment:
This statement is misleading. I'm comfortable claiming that `STRUCT<key ANY TYPE, value ANY TYPE>` is a BigQuery _convention_ for map types since they implement those particular field names for Avro input. But the `key_value` name for the struct is an artifact of Mozilla's own code for pre-processing Parquet datasets for import to BQ. We should try to be super-clear about the places where our functions are making assumptions about the structure of the data.

Reply:
how about this?
```suggestion
The BigQuery type that results from loading a parquet map is STRUCT<key_value ARRAY<STRUCT<key ANY TYPE, value ANY TYPE>>>.
```

Reply:
Looks good.